### PR TITLE
Control the number of rows in chunks returned by LimitExec

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Control the number of rows in chunks returned by `LimitExec`.
Following up #9293, this PR is a subtask of #9166.

### What is changed and how it works?
* Introduce `adjustRequiredRows` into `LimitExec` to adjust the required rows before each call of 
child's `Next`.
* Introduce `requiredRowsDataSource` as mock DataSource when test.
* Add some unit tests for `LimitExec`.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -769,6 +772,17 @@ func (e *LimitExec) Close() error {
 	return errors.Trace(e.baseExecutor.Close())
 }
 
+func (e *LimitExec) adjustRequiredRows(chk *chunk.Chunk) *chunk.Chunk {
```
Comment (zz-jason) [inline executor/executor.go]: please add some comments about this function to explain how the number of child required rows is determined.

Comment (qw4990) [reply executor/executor.go]: more comments have been added.

**Comment by qw4990** [comment]
PTAL~ @lysu

**Pull request merged**